test/e2e/pull_test.go

```diff
@@ -84,6 +84,36 @@ var _ = Describe("Podman pull", func() {
 		Expect(session).Should(Exit(0))
 	})
 
+	It("podman pull and run on split imagestore", func() {
+		tmpDir, err := os.MkdirTemp("", "")
+		Expect(err).ToNot(HaveOccurred())
+
+		// First clear all images
+		session := podmanTest.Podman([]string{"rmi", "--all"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		session = podmanTest.Podman([]string{"pull", "--imagestore", tmpDir, "quay.io/buildah/stable:latest"})
```

Comment:
Do not pull such big images in CI tests, this going to flake way to often.
I know nothing about the feature but couldn't we use podman load for this to avoid network flakes?

If you really have to pull then please use a tiny busybox image.

Reply:
Also split out the image name to a variable. One typo in your ToNot() check and it becomes a NOP.

Reply:
Done, switched to busybox.